src/main/java/macrobase/analysis/stats/mixture/NormalWishartClusters.java

```diff
@@ -106,12 +106,25 @@ public void initializeBaseForDP(List<Datum> data) {
      * Initializes atom (component) distributions. This method works great with DP mixture model.
      * @param data
      */
-    public void initializeAtomsForDP(List<Datum> data, Random random) {
+    public void initializeAtomsForDP(List<Datum> data, String filename, Random random) {
         omega = new ArrayList<>(K);
         dof = new double[K];
         beta = new double[K];
 
-        loc = BatchMixtureModel.gonzalezInitializeMixtureCenters(data, K, random);
+        if (filename != null) {
+            try {
+                loc = BatchMixtureModel.initializeClustersFromFile(filename, K);
+                log.debug("loc : {}", loc);
+                if(loc.size() < K) {
+                    loc = BatchMixtureModel.gonzalezInitializeMixtureCenters(loc, data, K, random);
+                }
+            } catch (FileNotFoundException e) {
+                e.printStackTrace();
```

Comment:
if file is provided, initialize K points from file, or if file contains less than K points in the file, take those points as seeds. I'll add this as a comment